lib/client/client_test.go

```diff
@@ -298,3 +300,38 @@ func (l wrappedListener) Accept() (net.Conn, error) {
 	close(l.acceptCh)
 	return l.Listener.Accept()
 }
+
+func TestLineLabeledWriter(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{
+			name:     "typical input",
+			input:    "this is\nsome test\ninput",
+			expected: "[label] this is\n[label] some test\n[label] input",
+		},
+		{
+			name:     "don't add empty line at end",
+			input:    "dangling newline\n",
+			expected: "[label] dangling newline\n",
+		},
+		{
+			name:     "blank lines in middle",
+			input:    "this\n\nis\n\nsome input",
+			expected: "[label] this\n[label] \n[label] is\n[label] \n[label] some input",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			buf := bytes.NewBuffer([]byte{})
```

Comment:
```suggestion
			var buf bytes.Buffer
```

`NewBuffer` is intended for cases when you need to control the underlying slice. In this case, you don't so the zero-valued buffer will work just fine.